unlock-app/src/services/web3Service.js

```diff
@@ -352,21 +352,23 @@ export default class Web3Service extends EventEmitter {
       return this.web3.eth
         .getTransactionReceipt(transactionHash)
         .then(transactionReceipt => {
-          // NOTE: old version of web3.js (pre 1.0.0-beta.34) are not parsing 0x0 into a falsy value
-          if (
-            !transactionReceipt.status ||
-            transactionReceipt.status == '0x0'
-          ) {
-            return this.emit('transaction.updated', transactionHash, {
-              status: 'failed',
-            })
+          if (transactionReceipt) {
+            // NOTE: old version of web3.js (pre 1.0.0-beta.34) are not parsing 0x0 into a falsy value
+            if (
+              !transactionReceipt.status ||
+              transactionReceipt.status == '0x0'
+            ) {
+              return this.emit('transaction.updated', transactionHash, {
+                status: 'failed',
+              })
+            }
+
+            return this.parseTransactionLogsFromReceipt(
```

Comment:
This is a stylistic thing, but it may be worthwhile to adjust this so that we explicitly check for the positive case and have the failures be the fall through 

`transactionReceipt && transactionReceipt.status &&  transactionReceipt.status !== '0x0' `

Reply:
Ho but this is a different thing right? `transactionReceipt.status && transactionReceipt.status !== '0x0' ` means the transaction actually failed...